pkg/common/upgrade/upgrade.go

```diff
@@ -78,6 +78,12 @@ func RunUpgrade() error {
 	done = false
 	if err = wait.PollImmediate(10*time.Second, MaxDuration, func() (bool, error) {
 		if viper.GetBool(config.Upgrade.ManagedUpgrade) && viper.GetBool(config.Upgrade.WaitForWorkersToManagedUpgrade) {
+			// Keep the managed upgrade's configuration overrides in place, in case Hive has replaced them
+			err = overrideOperatorConfig(h)
```

Comment:
As I understand it, the Hive sync would happen every two hours be default and the upgrade would take about an hour or so on an average. This change will certainly reduce the number of failures on the upgrade start itself due to sync override but is it also possible that the last Hive sync could be done 1.5 hours before an upgrade starts and within half on hour of upgrade the Hive sync takes place again to cause the same issue again?

Reply:
This syncs the configmap every 10 seconds. If hive has synced it, the change will be replaced in under that time period.

Reply:
Oh yes, ok understood. Thanks Matt !